scripts/setup-osx

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+########################################################################
+# This install script will setup your development dependencies on OS X.
+# Windows/Linux support is welcome!
+#
+# Usage: scripts/setup-osx.sh
+########################################################################
+
+_current_dir=$(cd "${BASH_SOURCE%/*}" && pwd)
+source "$_current_dir/lib/setup-osx/path-support.sh"
+
+source_lib "output.sh"
+source_lib "packages.sh"
+source_lib "platform.sh"
+source_lib "installers.sh"
+
+exit_unless_mac
+
+load_nvm_if_available
+load_rvm_if_available
+
+####
+setup_header "Installing requirements..."
+
+install_homebrew_if_needed
+brew_tap "caskroom/cask"
+
+brew_cask_install "caskroom/versions/java8"
+brew_install leiningen
+
+install_node
+brew_install watchman
+install_react_native_cli
+
+brew_cask_install android-sdk
+brew_install maven
+
+install_cocoapods
+
+####
+echo && setup_header "Installing dependencies..."
+
+dependency_setup lein deps
+dependency_setup npm install
+dependency_setup ./re-natal deps
```

Comment:
@dbalatero What is the difference between npm install and re-natal deps? I think they do the same (https://github.com/drapanjanas/re-natal/blob/master/re-natal.coffee#L144), and re-natal tries to use yarn if you have it installed, which I don't think it is ideal if the project doesn't use it yet. What do you think?

Reply:
Hey @pablodip! Totally agree that isn't ideal. re-natal _should_ check for `yarn.lock` before making those assumptions imho!

I'm ok with just doing `./re-natal deps`, so we should probably optimize for that.

I'm trying to get #2649 merged in, which will switch over to Yarn and make re-natal's weirdness a moot point.

Also I noticed that `@yarnpkg/lockfile` is marked as a dependency in the `develop` branch, which feels like a possible `git add . && git commit -m` goof? I wouldn't expect that package to be in the package-lock.json right now unless we were on yarn.

https://github.com/dbalatero/status-react/blob/c8168e9ff004e0bf523e27e039317b344030d9bf/package-lock.json#L7-L12

Anyways I think we should get #2649 in if it works out well, as it will speed up JS package installs significantly and has a more stable lock format in my experience. After that we can probably just switch to running `./re-natal deps` if it's already running `yarn install/npm install` like you said!

Reply:
Thanks for replying @dbalatero! :)

Just doing `npm|yarn install` could be more explicit, fix the current bug, and depend less on external tooks. I am okay with waiting for that PR so long as it is ready soon. Otherwise it may be interesting to fix this issue even with npm since both the setup and run-osx scrips are broken for people who have installed yarn, which I believe is quite common.

As for `@yarnpkg/lockfile`, really no idea right now. :S

Reply:
@pablodip wait wait, is all that `./re-natal deps` does check if versions are ok (which the lock file should be used for in npm or yarn's case) and then run `npm/yarn install`? I'm currently observing this behavior.

If so we can just get rid of running that altogether, and just run the correct npm/yarn command, which I think is what you're suggesting :)

Reply:
@pablodip Ok, I've removed that line on #2649's PR: https://github.com/status-im/status-react/pull/2649/files#diff-232884125f8738fe4f957f81e7915218L48

I'm just waiting to get that approved and merged. If you want to take any conversation over there to chat and/or encourage it to get merged that's great! Also if you have access to the private CI server, I need to know if my branch is passing or not heh.

Reply:
@dbalatero 

What it does is just installing npm/yarn I believe: https://github.com/drapanjanas/re-natal/blob/master/re-natal.coffee#L952

Remember to remove that command on the `run-osx` script as well, please. :) (https://github.com/status-im/status-react/blob/develop/scripts/run-osx#L64)

I don't have access to the CI yet, and some people are probably on holidays. Will let them know though. :)